applications/vessel-history/src/features/user/user.hooks.ts

```diff
@@ -28,5 +42,16 @@ export const useUser = () => {
     dispatch(logoutUserThunk({ redirectTo: 'home' }))
   }, [dispatch])
 
-  return { loading, logged, user, authorized, login, logout }
+  useEffect(() => {
+    if (!logged && (token || refreshToken || accessToken)) dispatch(fetchUserThunk())
```

Comment:
in other apps the token is refreshed automatically when the first request return unauthorised but this way you ensure you are always logged